source/state_representation/include/state_representation/trajectory/TrajectoryBase.hpp

```diff
@@ -453,14 +412,8 @@ template<typename StateT>
   requires std::derived_from<StateT, typename state_representation::State>
 inline void TrajectoryBase<TrajectoryT>::assert_not_contains_empty_state(const std::vector<StateT>& states) const {
   if (std::ranges::any_of(states, [&](const auto& state) { return state.is_empty(); })) {
+    // A bit unfortunate that we don't know the name of the state that was empty
     throw exceptions::EmptyStateException("Empty state variable provided");
```

Comment:
```suggestion
  if (std::string name; std::ranges::any_of(states, [&](const auto& state) { 
     name = state.get_name();
     return state.is_empty(); 
 })) {
    // A bit unfortunate that we don't know the name of the state that was empty
    throw exceptions::EmptyStateException("Empty state '" + name + "' variable provided");
```

something like this should work